sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkStringTypeUtils.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import org.apache.spark.sql.internal.SqlApiConf
+
+trait SparkStringTypeUtils {
+  /**
+   * Check if default collation is up-to-date with a config.
+   */
+  private def refreshDefaultCollation(): Unit = {
+    if (SqlApiConf.get.defaultCollation != DEFAULT_COLLATION) {
```

Comment:
Again, this might be slow.
1) @cloud-fan / @MaxGekk - is this the right way to react on config updates?
2) If we keep it like this, it would be good to make sure that `isDefaultCollation` is never called during query execution phase since this will likely be too slow. Maybe we can even add - `AnalysisHelper.inAnalyzer` check here?